server/service/handler.go

```diff
@@ -536,6 +536,10 @@ func attachFleetAPIRoutes(r *mux.Router, svc fleet.Service, config config.FleetC
 		errorLimiter.Limit("post_device_rotate_encryption_key", desktopQuota),
 	).POST("/api/_version_/fleet/device/{token}/rotate_encryption_key", rotateEncryptionKeyEndpoint, rotateEncryptionKeyRequest{})
 
+	demdm.WithCustomMiddleware(
+		errorLimiter.Limit("post_device_migrate_mdm", desktopQuota),
+	).POST("/api/_version_/fleet/device/{token}/migrate_mdm", migrateMDMDeviceEndpoint, deviceMigrateMDMRequest{})
```

Comment:
I think the endpoints with MDM configured middleware have to be added to `service.mdmAppleConfigurationRequiredEndpoints` (which tests that they all respond with the MDM not configured error).

Reply:
I looked at this initially and wasn't sure the best approach. The same list returned by the function gets tested in both Free and Premium. But for this particular endpoint, we'd expect there to be a license error on Premium and not configured on Free. I opted to leave it off the list and just added specific tests to each for [core](https://github.com/fleetdm/fleet/pull/11724/files#diff-c868f27ccf2a87facf88b1f46907dd7646bf27c18fa2f3ffdab56caa5b915052R4535) and [ee](https://github.com/fleetdm/fleet/pull/11724/files#diff-bbd0c5a6bc2f9a24e633031d4c6a3f5b0be7cbfe78ef1b56cdf9a7a2c32e21e2R2221). 

For the benefit of future us, I'll also add it to the list and just conditionally switch inside the test if it is this particular endpoint. That way future us won't get suckered into relying on a list that appears exhaustive but isn't. Thanks for the nudge in that direction!

Reply:
Ah gotcha, yeah maybe the list should evolve in the future to map the endpoint to an indication if it is premium-only or not, as I assume we'll have more and more premium-only endpoints that require mdm, and in that case the tests would be able to do the right thing automatically. (what you propose is fine, that's just thinking out loud for the future, we can make those changes when the next premium-only endpoint gets added)